zerver/lib/message.py

```diff
@@ -1102,6 +1106,8 @@ def extract_unread_data_from_um_rows(
     unmuted_stream_msgs: Set[int] = set()
     huddle_dict: Dict[int, RawUnreadHuddleDict] = {}
     mentions: Set[int] = set()
+    mentions_me_directly: Set[int] = set()
+    wildcard_mentions: Set[int] = set()
     total_unreads = 0
 
     raw_unread_messages: RawUnreadMessagesResult = dict(
```

Comment:
the reason we're getting `KeyError` downstream when calling `if message_id in raw_unread_data["mentions_me_directly"]` is because, even tho we initialized the sets that track `mentions_me_directly` and `wildcard_mentions`, we never actually added them to the `RawUnreadMessagesResult` that gets returned by this function (meaning the key is never populated, resulting in a key error when we attempt to pull from it)

```suggestion
    raw_unread_messages: RawUnreadMessagesResult = dict(
        mentions_me_directly=mentions_me_directly,
        wildcard_mentions=wildcard_mentions
```